tools/pipelines/test-perf-benchmarks.yml

```
@@ -70,160 +85,311 @@ stages:
+                mkdir -p ${{ variables.testWorkspace }}/node_modules/${{ testPackage }}/dist/test
+                tar -xvf $(Pipeline.Workspace)/test-files/$TAR_FILE.test-files.tar -C $(Pipeline.Workspace)/test-files
+                mv $(Pipeline.Workspace)/test-files/src/test/* ${{ variables.testWorkspace }}/node_modules/${{ testPackage }}/src/test
+                mv $(Pipeline.Workspace)/test-files/dist/test/* ${{ variables.testWorkspace }}/node_modules/${{ testPackage }}/dist/test
+
+          - task: Bash@3
+            displayName: 'Run memory performance test'
+            inputs:
+              targetType: 'inline'
+              workingDirectory: ${{ variables.testWorkspace }}/node_modules/${{ testPackage }}
+              script: |
+                echo "Run memory performance test for ${{ testPackage }}"
+                cp ${{ variables.testWorkspace }}/.npmrc . ;
+                npm i ;
+                npm run test:memory-profiling:report;
+
+          - task: CopyFiles@2
+            inputs:
+              sourceFolder: ${{ variables.testWorkspace }}/node_modules/${{ testPackage }}/.memoryTestsOutput
+              contents: '**'
+              targetFolder: ${{ variables.testWorkspace }}/memoryTestsOutput/${{ replace(replace(replace(testPackage, '@fluidframework/', '' ), '@fluid-', '' ), '/', '-') }}
+
+          - task: Bash@3
+            displayName: 'Cleanup package'
+            inputs:
+              targetType: 'inline'
+              workingDirectory: ${{ variables.testWorkspace }}/node_modules/
+              script: |
+                echo "Cleanup package ${{ testPackage }} from ${{ variables.testWorkspace }}/node_modules/"
+                rm -rf ${{ testPackage }};
 
         - task: Bash@3
           displayName: 'Write measurements to Aria/Kusto - memory usage'
           inputs:
             targetType: 'inline'
             workingDirectory: $(toolAbsolutePath)
             script: |
-              npm i;
-              npm run build:compile;
-              node --require @ff-internal/aria-logger bin/run --handlerModule $(toolAbsolutePath)/dist/handlers/memoryUsageTestHandler.js --dir '../../memoryProfilingOutput';
+              echo "Write the following benchmark output to Aria/Kusto"
+              ls -la ${{ variables.testWorkspace }}/memoryTestsOutput;
+              node --require @ff-internal/aria-logger bin/run --handlerModule $(toolAbsolutePath)/dist/handlers/memoryUsageTestHandler.js --dir ${{ variables.testWorkspace }}/memoryTestsOutput;
 
         - task: PublishPipelineArtifact@1
           displayName: 'Publish Artifact - Perf tests output - memory usage'
           inputs:
-            targetPath: '${{ variables.buildDirectory }}/memoryProfilingOutput'
+            targetPath: '${{ variables.testWorkspace }}/memoryTestsOutput'
             artifactName: 'perf-test-outputs_memory-usage'
-          condition: succeededOrFailed()
+            condition: succeededOrFailed()
```

Yep, once more here.
```suggestion
          condition: succeededOrFailed()
```